### PR TITLE
Put the user's manual and stdlib sample code under the CC0 license.

### LINKED ISSUE #12095: Consider licensing the Stdlib sample code under public domain/CC0
For users it's nice to be able to cut and paste sample code and remix it the way they see it fit without being encumbered by licensing duties. 

One way of doing this is to license the sample code present in Stdlib reference docs under a public domain/[CC0](https://creativecommons.org/share-your-work/public-domain/cc0/) license.

If there's agreement that this is a good thing then a blurb could be added [here](https://github.com/ocaml/ocaml/blob/trunk/manual/src/library/stdlib-blurb.etex) in the manual. Something like "All the sample code found in the reference documentation is in the public domain or available under a CC0 license".

(Note I'm not talking about the code in the tutorials, though that could be considered as well).

### PR DESCRIPTION
**Pull request by dbuenzli**
Closes #12095.

### CODE REVIEW

File: manual/src/foreword.etex [hunk 1]
```diff
@@ -54,13 +54,24 @@ Automatique (INRIA).
...
+The sample code in the user's manual and in the reference documentation
+of the standard library is licensed under a Creative Commons
+CC0 1.0 Universal (CC0 1.0) Public Domain Dedication License,
+\url{https://creativecommons.org/publicdomain/zero/1.0/}.
 \end{latexonly}
 
 \begin{htmlonly}
 \begin{rawhtml}
-<a id="cc_license_logo" rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"></a>
+<a id="cc_license_logo" rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://licensebuttons.net/l/by-sa/4.0/88x31.png"></a>
```
Comment (dbuenzli) [inline manual/src/foreword.etex:66]: Note the URL was changed here because that's where it redirects.

**Comment by xavierleroy** [comment]
The HTML version was poorly formatted, so I took the liberty to push an improved version.  This is what it looks like now:

image

**Review by xavierleroy — APPROVED** [review]
Looks good to me, thanks.  A lawyer might object that "the sample code" is not well defined, but I think it's good enough.  If anyone has a better wording to offer, please say so.

Also, since this is about IP, I'd like to have another approval, preferably by someone who is both an Inria employee and a representative of the OCSF, meaning @gasche or @damiendoligez .

**Comment by gasche** [comment]
Do I correctly understand that the manual examples are covered by the CLA, and that this is the reason why we can easily relicense them without the usual dance of contacting all authors to get approval?

**Comment by avsm** [comment]
The manual claims [copyright is owned by INRIA](https://github.com/ocaml/ocaml/pull/12097/files#diff-f20c36be9a9045c0833f2babe77dd1025b04ed664f7bbdc5f4999bc8af2c2c27R48) and not by the individual authors, so this is presumably the CLA in action.

File: Changes [hunk 2]
```diff
@@ -147,6 +147,10 @@ Working version
 
 ### Standard library:
 
+- 12095: Put the sample code of the user's manual and reference
+  documentation of the standard library under the CC0 1.0 Universal
+  (CC0 1.0) Public Domain Dedication license.
```
Comment (gasche) [inline Changes]: This should go in the "Manual and documentation" section I suppose.

File: manual/src/foreword.etex (hunk 1 above)
Comment (gasche) [reply manual/src/foreword.etex:66]: Is it a good idea still? The first URL looks more official-looking and may be more stable? (No strong opinion.)

**Review by gasche — APPROVED** [review]
Thanks for the clarification. With this in mind I am happy to approve of the license clarification, which I think is more convenient for everyone.

File: manual/src/foreword.etex (hunk 1 above)
Comment (dbuenzli) [reply manual/src/foreword.etex:66]: Yes, it's a 301. That means update your links.

```
> curl -s -o /dev/null -w "%{http_code}" https://i.creativecommons.org/l/by-sa/4.0/88x31.png
301
```

**Comment by xavierleroy** [comment]
> The manual claims [copyright is owned by INRIA](https://github.com/ocaml/ocaml/pull/12097/files#diff-f20c36be9a9045c0833f2babe77dd1025b04ed664f7bbdc5f4999bc8af2c2c27R48) and not by the individual authors

That's actually over-simplified :-(  The manual is mostly copyright Inria, but there are substantial contributed parts that are copyright by their authors, but but we have CLAs giving Inria permission to distribute *and relicense* those parts as it sees fit.

> so this is presumably the CLA in action.

The rights to relicense is the CLA in action, yes.

Also note that very small code examples probably generate no copyrights, as they fall under the "Works consisting entirely of information that is common property and containing no original authorship" category.

**Comment by gasche** [comment]
Yes, but for example the effect-handler examples are interesting and non-trivial, I think they would definitely require author agreement in absence of a CLA -- if we wanted to do things very carefully, etc.

**Comment by xavierleroy** [comment]
> the effect-handler examples are interesting and non-trivial, I think they would definitely require author agreement in absence of a CLA

I agree with this, but fortunately we have a CLA from the author.  I think we've been fairly consistent so far on the policy small contribution = no copyright, significant contribution = a CLA.

**Pull request merged**